pkg/docker/docker.go

```diff
@@ -823,6 +952,9 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 			c.Close()
 		}
 
+		if attachError != nil {
+			return attachError
+		}
```

Comment:
shouldn't we be looking for this earlier?  (and don't we need to ensure the attach has run before checking it?  and since we're setting it in a go routine i'm guessing we need a memory barrier to ensure we're seeing the updated value)